### PR TITLE
Addressing previous code review and test session comments

### PR DESCRIPTION
**Pull request by panasetskaya**
Addressing comments from previous code reviews + app test session comments here.

See my comments below :)

There is a known edge case that happens when we already made a change in tier/deductible, but it's not activated yet, and then we try to go through the flow again. Then the ui breaks a bit in one place. But I'm going to talk it over with BE, maybe it won't even be allowed - to go through flow again while the first change is not activated yet.

### CODE REVIEW

File: app/feature/feature-choose-tier/src/main/kotlin/com/hedvig/android/feature/change/tier/ui/stepcustomize/SelectTierDestination.kt [hunk 1]
```diff
@@ -197,6 +206,8 @@ private fun SelectTierScreen(
   onChooseTierClick: () -> Unit,
   onChooseDeductibleInDialogClick: (quote: TierDeductibleQuote) -> Unit,
   onChooseTierInDialogClick: (tier: Tier) -> Unit,
+  onSetDeductibleBackToPreviouslyChosen: () -> Unit,
+  onSetTierBackToPreviouslyChosen: () -> Unit,
```
Comment (panasetskaya) [inline app/feature/feature-choose-tier/src/main/kotlin/com/hedvig/android/feature/change/tier/ui/stepcustomize/SelectTierDestination.kt:210]: So I think here, changes in this screen  (and its ViewModel) are mostly what needs to be reviewed again just a little bit. Otherwise it's just small fixes according to the previous comments.
What I'm trying to achieve here, is that we would be able to press cancel (or outside of the dialog), and the choice locally made in dialog would be dismissed, and we would go back to what was previously chosen. That's why I'm storing two choices - "global" and "dialog local".
I tried not to pass local choice to the viewModel at all, but it behaved strangely sometimes then, so everything goes through vm now.

File: app/compose/compose-ui/src/main/kotlin/com/hedvig/android/compose/ui/StringWithShiftedLabel.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.style.BaselineShift
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.unit.TextUnit
+
+fun stringWithShiftedLabel(
```
Comment (panasetskaya) [inline app/compose/compose-ui/src/main/kotlin/com/hedvig/android/compose/ui/StringWithShiftedLabel.kt:11]: extracted this, using it in 4 places now

File: app/feature/feature-choose-tier/src/main/kotlin/com/hedvig/android/feature/change/tier/ui/stepcustomize/SelectCoverageViewModel.kt [hunk 3]
```diff
@@ -82,23 +83,22 @@ private class SelectCoveragePresenter(
...
-//          locallyChosen?.let { local ->
-//            chosenTier = local
-//            // try to pre-choose a quote with the same deductible and newly chosen coverage
-//            // if there is no such quote, the deductible will not be per-chosen
-//            val previouslyChosenDeductible = chosenQuote?.deductible
-//            val quoteWithNewTierOldDeductible =
-//              state.map[local]!!.firstOrNull { it.deductible == previouslyChosenDeductible }
-//            chosenQuote = quoteWithNewTierOldDeductible
-//          }
+          // try to pre-choose a quote with the same deductible and newly chosen coverage
```
[the remark this answers is not in the document]
Comment (panasetskaya) [reply app/feature/feature-choose-tier/src/main/kotlin/com/hedvig/android/feature/change/tier/ui/stepcustomize/SelectCoverageViewModel.kt:86]: Yeah, sure. I'll make a separate little PR for that

[the remark this answers is not in the document]
Comment (panasetskaya) [reply app/feature/feature-choose-tier/src/main/kotlin/com/hedvig/android/feature/change/tier/ui/stepcustomize/SelectCoverageViewModel.kt:85]: addressed in unit-testing branch
